### ISSUE TITLE
ExprLocationOf returns null with loop-values

### ISSUE THREAD

**Issue by TPGamesNL** [SkriptLang/Skript#2878]
### Description
The `location of` expression returns null in some scenarios.

### Steps to Reproduce
```
loop player and player:
	broadcast "L1 %location of loop-value%"
```
returns `<none>` twice

```
loop player and player:
	set {_p} to loop-value
	broadcast "L2 %location of {_p}%"
```
works correctly

```
set {_p::*} to player and player
loop {_p::*}:
	broadcast "L3 %location of loop-value%"
```
works correctly

I have also tried with other expressions, including health and head location, but couldn't reproduce this bug with those expressions.

### Expected Behavior
The expression should return the actual location, instead of `<none>`.

### Errors / Screenshots
![image](https://user-images.githubusercontent.com/29547183/77460903-0caee580-6e02-11ea-9a7e-f6fce0760e2e.png)

### Server Information
* **Server version/platform:** 1.15.2 (Paper-143)
* **Skript version:** 2.5-alpha2

**Comment by Jayderp** [comment]
Could be due to the alpha having buggy bug bugs

**Comment by Jayderp** [comment]
Use [icode]location of loop-value[/icode] instead

**Comment by TPGamesNL** [comment]
> Use `location of loop-value` instead

I'm using that already?

**Comment by Jayderp** [comment]
> > Use `location of loop-value` instead
> 
> I'm using that already?

If you're using that I don't understand the point of using `location of {_p}` as it's simpler to write 1 line instead of 2 or 3

**Comment by legofreak1999** [comment]
> > > Use `location of loop-value` instead
> > 
> > 
> > I'm using that already?
> 
> If you're using that I don't understand the point of using `location of {_p}` as it's simpler to write 1 line instead of 2 or 3

Did u even look what the bug is about?

location of loop-value is the whole bug? its not about writing something simpler? Its purely a bug?

**Comment by TPGamesNL** [comment]
Skript chooses the wrong converter on parsetime, as it chooses a `Block` -> `Location` converter for `ExprLoopValue` in `ExprLocationOf`. The problem lies with either of two things (or both):
1. [ScriptLoader.java#L1051](https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/ScriptLoader.java#L1051) changes the return type of the looped expression to `Object`, which changes the return type of `ExprLoopValue` to `Object`, which means that the converter will be an `Object` -> `Location` converter, instead of a `Player -> Location` converter, which is what we want.
2. [Converters.java#L315](https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/registrations/Converters.java#L315), since converting from an `Object` should in almost all cases be done with a generic converter (that determines the correct converter on runtime when the class is actually know), not just by choosing a random converter. `Converters.getConverter(Object.class, Location.class)` currently returns a `Block` -> `Location` converter.

**Comment by TPGamesNL** [comment]
Fixed in #4020 (from comment above, the first fix was applied)
